### PR TITLE
Add `domainName` property to API Gateway V1 event

### PR DESCRIPTION
**Pull request by MPLew-is**
Add support for the existing `domainName` property to API Gateway V1 events in Swift.

### Motivation:

According to [AWS example events](https://github.com/awsdocs/aws-lambda-developer-guide/blob/cad20e9f298fa16edfe0fb643ab8118b0fe2dc54/sample-apps/nodejs-apig/event.json) and documentation, this seems to be included in all requests.
We should be able to access this seamlessly in Swift.

### Modifications:

- Added `domainName` property to `requestContext` in `APIGatewayRequest` object
- Added example values for `domainName` to corresponding tests

### Result:

After this change, users can access the domain name with:
```swift
struct ApiGatewayHandler: LambdaHandler {
	typealias Event  = APIGatewayRequest
	typealias Output = APIGatewayResponse

	func handle(_ event: Event, context: LambdaContext) async throws -> Output {
		...
		let domainName: String = event.domainName
		...
	}
}
```

### CODE REVIEW

File: Sources/AWSLambdaEvents/APIGateway.swift [hunk 1]
```diff
@@ -37,6 +37,7 @@ public struct APIGatewayRequest: Codable {
 
         public let resourceId: String
         public let apiId: String
+        public let domainName: String
```
Comment (tomerd) [inline Sources/AWSLambdaEvents/APIGateway.swift:40]: should this be optional?

**Comment by tomerd** [comment]
@swift-server-bot test this please

File: Sources/AWSLambdaEvents/APIGateway.swift (hunk 1 above)
Comment (MPLew-is) [reply Sources/AWSLambdaEvents/APIGateway.swift:40]: I have yet to find a request that doesn't include this field, but I'm happy to make it optional if that would reduce potential breakages under corner cases I haven't seen yet.

Comment (MPLew-is) [reply Sources/AWSLambdaEvents/APIGateway.swift]: Done

**Review by tomerd — APPROVED** [review]

**Pull request merged**